server/models/default_local_provider.go

```diff
@@ -1009,24 +1080,44 @@ func (l *DefaultLocalProvider) GetConnectionsStatus(_ *http.Request, _ string) (
 	return nil, ErrLocalProviderSupport
 }
 
-func (l *DefaultLocalProvider) UpdateConnection(_ *http.Request, _ *connections.Connection) (*connections.Connection, error) {
-	return nil, ErrLocalProviderSupport
+func (l *DefaultLocalProvider) UpdateConnection(_ *http.Request, connection *connections.Connection) (*connections.Connection, error) {
+	return l.ConnectionPersister.UpdateConnectionByID(connection)
 }
 
 func (l *DefaultLocalProvider) UpdateConnectionStatusByID(token string, connectionID uuid.UUID, connectionStatus connections.ConnectionStatus) (*connections.Connection, int, error) {
-	return nil, http.StatusForbidden, ErrLocalProviderSupport
+	updatedConnection, err := l.ConnectionPersister.UpdateConnectionStatusByID(connectionID, connectionStatus)
+	if err != nil {
+		fmt.Println("on update connection error", err)
+		return nil, http.StatusInternalServerError, err
+	}
+	return updatedConnection, http.StatusOK, nil
 }
 
-func (l *DefaultLocalProvider) UpdateConnectionById(_ *http.Request, _ *ConnectionPayload, _ string) (*connections.Connection, error) {
-	return nil, ErrLocalProviderSupport
+func (l *DefaultLocalProvider) UpdateConnectionById(req *http.Request, conn *ConnectionPayload, _ string) (*connections.Connection, error) {
```

Comment:
Are we being schema-driven here? No schema reference and specific version?

Reply:
No we are not, we need to define schema for connections, I have checked meshery/schemas repo, the schema which is present there is API schema for connection not the connection schema